src/Main.hs

```diff
@@ -189,33 +173,35 @@ splitBehaviour store (Transition name contract iface@(Interface _ decls) iffs' c
       storageBounds = snd $ getStorageBounds env
   iff <- checkIffs env (iffs' <> calldataBounds <> storageBounds)
   postcondition <- mapM (checkBool env) (fromMaybe [] maybePost)
-  flatten iff postcondition [] cases
+  flatten iff postcondition cases
   where
     env = mkEnv contract store decls
 
     -- translate wildcards into negation of other cases
-    normalize :: [Case Expr b] -> [Case Expr b]
-    normalize =
-      snd
-        . mapAccumL
-          ( \a b -> case b of
-              Leaf pn WildExp p -> (a, Leaf pn (ENot nowhere (andRaw a)) p)
-              Branch pn WildExp p -> (a, Branch pn (ENot nowhere (andRaw a)) p)
-              e@(Leaf _ c _) -> (c : a, e)
-              e@(Branch _ c _) -> (c : a, e)
-          )
-          []
-
-    -- flatten case tree
-    flatten :: [Exp Bool] -> [Exp Bool] -> [Exp Bool] -> Case Expr Post -> Err [Claim]
-    flatten iff postc pathcond (Leaf _ cond post) = do
-      c <- checkBool env cond
-      (p, maybeReturn, contracts) <- checkPost env contract post
-      return $ splitCase name False contract iface (c : pathcond) iff maybeReturn p postc contracts
-    flatten iff postc pathcond (Branch _ cond cs) = do
-      c <- checkBool env cond
-      leaves <- mapM (flatten iff postc (c : pathcond)) (normalize cs)
-      return $ join leaves
+    normalize :: [Case] -> [Case]
+    normalize cases = snd $ mapAccumL checkCase (BoolLit False) cases
+
+    checkCase :: Expr -> Case -> (Expr, Case)
+    checkCase acc (Case p WildExp post) =
+      (error "wildcard not last case",
+        Case p (ENot nowhere acc) post)
+    checkCase acc (Case p e post) = (EOr nowhere e acc, Case p e post)
```

Comment:
I don't really understand what's going on here. Won't this fail for any list of cases that has a wildcard in it somewhere?

Reply:
My intention was to ensure that you can only have wildcard in the last case, as otherwise that case would be redundant. My thinking here is that we only refer to the first element of this tuple if we keep iterating through the array; otherwise we just consider the second element. But this is a bit of an ugly thing (not proper error handling), this is part of what makes this PR a draft, I suppose